src/web_app/src/pages/drl_hyperParams.tsx

```diff
@@ -0,0 +1,210 @@
+import logging from "../config/logging";
+import {useNavigate, useLocation} from "react-router-dom";
+import Typography from "@mui/material/Typography";
+import Grid from "@mui/material/Grid";
+import TextField from "@mui/material/TextField";
+import Stack from "@mui/material/Stack";
+import Button from "@mui/material/Button";
+import {Types} from "../types/data";
+import axios from "axios";
+import Backdrop from "@mui/material/Backdrop";
+import sleep from "../config/sleep_funct";
+import CircularProgress from "@mui/material/CircularProgress";
+import FormControlLabel from "@mui/material/FormControlLabel";
+import Checkbox from "@mui/material/Checkbox";
+
+export function DrlHyperParamsPage() {
+    const url = useLocation().pathname;
+    const dataset_choice = url.split("/")[2];
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+    const navigate = useNavigate();
+    const [
+        open,
+        setOpen
+    ] = React.useState(false);
+    const [
+        override_check,
+        setOverride_check
+    ] = React.useState(false);
+    const [
+        nb_epochs,
+        setNb_epochs
+    ] = React.useState<number>(10000);
+    const [
+        learning_rate,
+        setLearning_rate
+    ] = React.useState<number>(0.9);
+    const [
+        batch_size,
+        setBatch_size
+    ] = React.useState<number>(32);
+    const [
+        momemtum,
+        setMomentum
+    ] = React.useState<number>(0.2);
+    const [
+        param,
+        setParam
+    ] = React.useState<Types.Hyper_parameters>({
+        type  : "drl",
+        params: {
+            "nb_epochs"    : 1000,
+            "pop_size"     : 512,
+            "seed"         : 0.5,
+            "mutation_rate": 0.2,
+            "learning_rate": 0.9,
+            "batch_size"   : 32,
+            "momentum"     : 0.2
+        },
+        override     : false,
+        bench_id     : "nothing",
+        snapshot_rate: 3
+    });
+
+    useEffect(() => {
+        logging.info(`Loading ${url}`);
+    });
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    const handleClickNext = async() => {
+        setParam({
+            "type"  : "drl",
+            "params": {
+                "nb_epochs"    : nb_epochs,
+                "pop_size"     : 512,
+                "seed"         : 0.5,
+                "mutation_rate": 0.2,
+                "learning_rate": learning_rate,
+                "batch_size"   : batch_size,
+                "momentum"     : momemtum
+            },
+            override     : override_check,
+            bench_id     : "nothing",
+            snapshot_rate: 3
+        });
+
+        axios.post("http://127.0.0.1:5000/operation_params/drl", {
+            "d_c"         : dataset_choice,
+            "hyper_params": JSON.stringify(param)
+        }, {headers: {"Content-Type": "multipart/form-data"}})
+            .then(async() => {
+                setOpen(true);
+                await sleep(5000);
```

Comment:
Ah! I just want to be sure that the data was sent well before continuing with the rest that's why I put it in the then. In case, there is a problem i will add a snackbar to notice the user there is a problem.

Reply:
Ok I will decrement the time

Reply:
> I just want to be sure that the data was sent well before continuing with the rest that's why I put it in the then

As long as you are in the `then` then the data has been sent, processed by the server and the server answered.
That's the goal of async requests.